src/Security/src/Authentication.CloudFoundryCore/MutualTlsAuthenticationOptionsPostConfigurer.cs

```diff
@@ -2,24 +2,50 @@
 // The .NET Foundation licenses this file to you under the Apache 2.0 License.
 // See the LICENSE file in the project root for more information.
 
+using Microsoft.AspNetCore.Authentication.Certificate;
+using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
 using Steeltoe.Common.Options;
 using Steeltoe.Security.Authentication.Mtls;
+using System.Collections.Generic;
+using System.Security.Claims;
+using System.Threading.Tasks;
 
 namespace Steeltoe.Security.Authentication.CloudFoundry
 {
     public class MutualTlsAuthenticationOptionsPostConfigurer : IPostConfigureOptions<MutualTlsAuthenticationOptions>
     {
         private readonly IOptionsMonitor<CertificateOptions> _containerIdentityOptions;
+        private readonly ILogger<CloudFoundryInstanceCertificate> _logger;
 
-        public MutualTlsAuthenticationOptionsPostConfigurer(IOptionsMonitor<CertificateOptions> containerIdentityOptions)
+        public MutualTlsAuthenticationOptionsPostConfigurer(IOptionsMonitor<CertificateOptions> containerIdentityOptions, ILogger<CloudFoundryInstanceCertificate> logger = null)
         {
             _containerIdentityOptions = containerIdentityOptions;
+            _logger = logger;
         }
 
         public void PostConfigure(string name, MutualTlsAuthenticationOptions options)
         {
             options.IssuerChain = _containerIdentityOptions.CurrentValue.IssuerChain;
+            options.Events = new CertificateAuthenticationEvents()
+            {
+                OnCertificateValidated = context =>
+                {
+                    var claims = new List<Claim>(context.Principal.Claims);
+                    if (CloudFoundryInstanceCertificate.TryParse(context.ClientCertificate, out var cfCert, _logger))
```

Comment:
Tim, I did not see a test, but in running this did you see the log message showing up? I don't see why the logger wouldn't get resolved by the container, but just making sure.

Reply:
Yes, the log message does show up if there is a logger and cert validation fails